include/flash.h

```diff
@@ -107,13 +121,9 @@ static inline int flash_erase(struct device *dev, off_t offset, size_t size)
 /**
  *  @brief  Enable or disable write protection for a flash memory
  *
- *  This API is required to be called before the invocation of write or erase
- *  API. Please note that on some flash components, the write protection is
- *  automatically turned on again by the device after the completion of each
- *  write or erase calls. Therefore, on those flash parts, write protection needs
- *  to be disabled before each invocation of the write or erase API. Please refer
- *  to the sub-driver API or the data sheet of the flash component to get details
- *  on the write protection behavior.
```

Comment:
Are you sure that all existing flash drivers work in the way you've changed the documentation to? My concern is that we are creating "bugs" in drivers that are working according to the existing contract by changing it without fixing them.